php/libraries/Recording.class.inc

```diff
@@ -0,0 +1,425 @@
+                    array('v_fileid' => $fileid)
+                );
+            if (empty($dbrow)) {
+                throw new \NotFound('There is no file with that FileID');
+            }
+            $this->_fileid       = $fileid;
+            $this->_filename     = basename($dbrow['filelocation']);
+            $this->_filelocation = $dbrow['filelocation'];
+            $this->_outputtype   = $dbrow['outputtype'];
+            $this->_acquisitionmodality = $dbrow['acquisitionmodality'];
+            $this->_filetype            = $dbrow['filetype'];
+            $this->_centerid            = (int) $dbrow['centerid'];
+            $this->_entitytype          = $dbrow['entitytype'];
+            $this->_channel_file_path   = $dbrow['channel_file_path'];
+            $this->_electrode_file_path = $dbrow['electrode_file_path'];
+            $this->_event_file_path     = $dbrow['event_file_path'];
+            $this->_archive_file_path   = $dbrow['archive_file_path'];
+        }
+    }
+
+    /**
+     * Get a specific file header from physiological_parameter file
+     *
+     * @param string $headername The parameter name to retrieve
+     *
+     * @return string The value or an empty string
+     */
+    public function getMetadata(string $headername): string
+    {
+        $dbrow = \NDB_Factory::singleton()
+            ->database()
+            ->pselectRow(
+                '
+                       SELECT
+                         Value
+                       FROM
+                         physiological_parameter_file pf 
+                       JOIN parameter_type pt
+                         USING (ParameterTypeID)
+                       JOIN physiological_file f
+                         USING (PhysiologicalFileID)
+                       WHERE
+                         f.PhysiologicalFileID = :v_fileid AND
+                         pt.Name = :v_headername
+                      ',
+                array(
+                    'v_fileid'     => $this->_fileid,
+                    'v_headername' => $headername,
+                )
+            );
+        return $dbrow['Value'] ?? '';
+    }
+
+    /**
+     * Get all metadata.class from the physiological_parameter files table for that
+     * Recording.
+     *
+     * @return array An associative array of {Name => Value}
+     */
+    public function getAllMetadata(): array
+    {
+        $dbrows = \NDB_Factory::singleton()
+            ->database()
+            ->pselect(
+                '
+                       SELECT
+                         pt.Name as name,
+                         pf.Value as value
+                       FROM
+                         physiological_parameter_file pf 
+                       JOIN parameter_type pt
+                         USING (ParameterTypeID)
+                       JOIN physiological_file f
+                         USING (PhysiologicalFileID)
+                       WHERE
+                         f.PhysiologicalFileID = :v_fileid
```

Comment:
same as before here. Maybe you would like to put `physiological_file` in a sub-query.

Reply:
Again no need to include ` physiological_file` in the query. A JOIN between 2 tables is enough.